### PR TITLE
Confirm overwrite workspace

### PR DESCRIPTION
**Pull request by honeybadgerofdoom**
This PR adds an alert asking users to confirm before they overwrite an existing workspace

### CODE REVIEW

**Review by pierce-smith1 — COMMENTED** [review]
Works:tm: beautifully, will approve once the CC issues are fixed. You can probably ignore the cognitive complexity one but the code duplication issues seem valid.
(On that note, why are we just now getting those two code duplication issues? Hasn't that code existed for a while)?

File: src/js/ui/NewDataExploration/Load.js [hunk 1]
```diff
@@ -54,6 +54,64 @@ You may add Your own copyright statement to Your modifications and may provide a
 
 9. Accepting Warranty or Additional Liability. While redistributing the Work or Derivative Works thereof, You may choose to offer, and charge a fee for, acceptance of support, warranty, indemnity, or other liability obligations and/or rights consistent with this License. However, in accepting such obligations, You may act only on Your own behalf and on Your sole responsibility, not on behalf of any other Contributor, and only if You agree to indemnify, defend, and hold each Contributor harmless for any liability incurred by, or claims asserted against, such Contributor by reason of your accepting any such warranty or additional liability. 
 
+END OF TERMS AND CONDITIONS
```
Comment (drg101) [inline src/js/ui/NewDataExploration/Load.js:57]: Extra licence was inserted here for some reason.

File: src/js/ui/NewDataExploration/Load.js [hunk 2]
```diff
@@ -101,7 +159,7 @@ export default React.memo(function Load({deSerializeWorkspace, setModalOpen}) {
                 <Typography className={classes.title} align="center" variant="h5">Load Workspace</Typography>
             </Grid>
             <Grid item className={classes.fullWidth}>
-                <SavedWorkspaceSlotSelection title="Select a Saved Workspace" slotCurrentlySelected={slotCurrentlySelected} setSlotCurrentlySelected={setSlotCurrentlySelected} onlyShowFullSlots/>
+                <SavedWorkspaceSlotSelection title="Select a Saved Workspace" setNameOfCurrentSlot={() => {}} slotCurrentlySelected={slotCurrentlySelected} setSlotCurrentlySelected={setSlotCurrentlySelected} onlyShowFullSlots/>
```
Comment (drg101) [inline src/js/ui/NewDataExploration/Load.js:162]: Why is setNameOfCurrentSlot just a empty function, should this be deleted?

**Review by drg101 — CHANGES_REQUESTED** [review]
A few things on the logic of how it is determined that a slot already has a name.
Right now, the `setNameOfCurrentSlot` is used to pass this up to the `Save` component. The `Save` component already has a reference to the current slot with `slotCurrentlySelected` though, so this other prop is redundant. 

Looks good other than this though, will approve once the state stuff is addressed.

File: src/js/ui/NewDataExploration/Load.js (hunk 2 above)
Comment (honeybadgerofdoom) [reply src/js/ui/NewDataExploration/Load.js:162]: It needs to be there or SavedWorkspaceSlotSelection throws an error when called from Load.js.

**Comment by honeybadgerofdoom** [comment]
> A few things on the logic of how it is determined that a slot already has a name.
> Right now, the `setNameOfCurrentSlot` is used to pass this up to the `Save` component. The `Save` component already has a reference to the current slot with `slotCurrentlySelected` though, so this other prop is redundant.
> 
> Looks good other than this though, will approve once the state stuff is addressed.

slotCurrentlySelected is an int. If I understand correctly it is used to index into a data structure that isn't visible in Save.js, so I created another piece of state to hold the name (or null) at that index. Not sure how I could use just this int to get the name of the selected slot, or if it is null.

**Comment by honeybadgerofdoom** [comment]
> A few things on the logic of how it is determined that a slot already has a name.
> Right now, the `setNameOfCurrentSlot` is used to pass this up to the `Save` component. The `Save` component already has a reference to the current slot with `slotCurrentlySelected` though, so this other prop is redundant.
> 
> Looks good other than this though, will approve once the state stuff is addressed.

K I'll work on this, thanks!

**Review by drg101 — CHANGES_REQUESTED** [review]
One more thing, could "Empty Slot" be changed to something that makes more sense like "Saved Workspace" or something?

**Pull request merged**